packages/rule-components/rollup.config.js

```diff
@@ -0,0 +1,92 @@
+/* eslint-disable camelcase */
+
+import nodeResolve from 'rollup-plugin-node-resolve';
+import commonjs from 'rollup-plugin-commonjs';
+import babel from 'rollup-plugin-babel';
+import nodeGlobals from 'rollup-plugin-node-globals';
+import { terser } from 'rollup-plugin-terser';
+import postcss from 'rollup-plugin-postcss';
+import json from '@rollup/plugin-json';
+import analyze from 'rollup-plugin-analyzer';
+import { dependencies, name } from './package.json';
+import { createFilter } from 'rollup-pluginutils';
+import glob from 'glob';
+
+const globMapper = (mapper) => glob.sync(mapper).map(item => {
+    const [ path ] = item.split('/index.js');
+    const last = path.split('/').pop();
+    if (last === 'src') {
+        return {
+            index: item
+        };
+    }
+
+    return {
+        [last]: item
+    };
+}).reduce((acc, curr) => ({ ...acc, ...curr }), {});
+
+const external = createFilter(
+    Object.keys(dependencies).map(item => item.includes('@patternfly') ? `${item}/**` : item),
+    null,
+    { resolve: false }
+);
+
+const globals = {
+    react: 'React',
+    'react-dom': 'ReactDOM',
+    '@patternfly/react-core': '@patternfly/react-core',
+    '@patternfly/react-icons': '@patternfly/react-icons'
+};
+
+const commonjsOptions = {
+    ignoreGlobal: true,
+    include: /node_modules/
+};
+
+const babelOptions = {
+    exclude: /node_modules/,
+    runtimeHelpers: true,
+    configFile: './babel.config'
+};
+
+const plugins = [
+    nodeResolve(),
+    babel(babelOptions),
+    commonjs(commonjsOptions),
+    nodeGlobals(),
+    terser({
+        keep_classnames: true,
+        keep_fnames: true
+    }),
+    postcss({
+        inject: true
+    }),
+    analyze({ summaryOnly: true }),
+    json()
+];
+
+export default [
+    ...[ 'cjs', 'esm' ].map(env => ({
+        input: globMapper('src/**/index.js'),
+        output: {
+            dir: `./dist/${env}`,
+            format: env,
+            name,
+            globals
+        },
+        external,
+        plugins
+    })),
+    ...Object.entries(globMapper('src/**/index.js')).map(([ key, input ]) => ({
+        input,
+        output: {
+            file: `./dist/umd/${key}.js`,
```

Comment:
I am not sure if it's a good idea to code a split UMD package. You can't read from any UMD/IIFE package so you will not avoid bundling all the necessary resources into a UMD package. You risk having the same code multiple times in multiple outputs.

Reply:
That's actually what is needed from umd output - multiple files with similiar content. If you end up using only 1/3 of the code you don't really need the rest of it.

Reply:
I was thinking more about you can end up with 10x the same code if you use 10 different files that all use the same common function for instance. The end result might be much worse than just having it all in one file. UMD is meant to be self-contained.

Unless the files/components themselves are supposed to be self-contained and wont reference each other in the builds.